src/scripts/create-user.ts

```diff
@@ -0,0 +1,133 @@
+import { CreateUserRequestOptions } from '@okta/okta-sdk-nodejs';
+import { Argv } from 'yargs';
+import { RootCommand } from '..';
+
+import {
+  oktaManageClient,
+  OktaConfiguration,
+  User,
+  oktaUserAsUser,
+  getUser,
+} from './services/user-service';
+
+import { generatePassword } from './services/password_manager';
+
+const createUser = async (
+  oktaConfiguration: OktaConfiguration,
+  password: string,
+  email: string,
+  firstName = 'Unknown',
+  lastName = 'User'
+): Promise<User> => {
+  const client = oktaManageClient(oktaConfiguration);
+
+  const maybeOktaUser = getUser(email, client);
+  // eslint-disable-next-line functional/no-expression-statement
+  const newUser: CreateUserRequestOptions = {
+    profile: {
+      firstName: firstName,
+      lastName: lastName,
+      email: email,
+      login: email,
+    },
+    credentials: {
+      password: { value: password },
+    },
+  };
+
+  // eslint-disable-next-line functional/functional-parameters
+  const throwOnExisting = () => {
+    // eslint-disable-next-line functional/no-throw-statement
+    throw new Error(
+      `User [${email}] already exists. Can not create a new user.`
+    );
+  };
+
+  return (await maybeOktaUser) === undefined
+    ? oktaUserAsUser(await client.createUser(newUser))
+    : throwOnExisting();
+};
+
+export default (
+  // eslint-disable-next-line @typescript-eslint/prefer-readonly-parameter-types
+  rootCommand: RootCommand
+): Argv<{
+  readonly clientId: string;
+  readonly privateKey: string;
+  readonly organisationUrl: string;
+  readonly email: string;
+  readonly firstName?: string;
+  readonly lastName?: string;
+}> =>
+  rootCommand.command(
+    'create-user [email]',
+    'Deletes the specified user',
+    // eslint-disable-next-line functional/no-return-void, @typescript-eslint/prefer-readonly-parameter-types
+    (yargs) => {
+      // eslint-disable-next-line functional/no-expression-statement
+      yargs
+        .option('email', {
+          type: 'string',
+          alias: ['login'],
+          // eslint-disable-next-line quotes
+          describe: "The new user's login/email",
+          demandOption: true,
+        })
+        .option('fname', {
+          type: 'string',
+          alias: ['first-name'],
+          // eslint-disable-next-line quotes
+          describe: "The new user's first name",
+          demandOption: true,
+        })
+        .option('lname', {
+          type: 'string',
+          alias: ['last-name'],
+          // eslint-disable-next-line quotes
+          describe: "The new user's last name",
+          demandOption: true,
+        });
+    },
+    async (args: {
+      readonly clientId: string;
+      readonly privateKey: string;
+      readonly organisationUrl: string;
+      readonly email: string;
+      readonly firstName?: string;
```

Comment:
`demandOption` is set to `true` for these. Why are they optional arguments?

Reply:
The optional parameters were legacy code, and it has now been updated to be required